src/main/java/io/github/externschool/planner/service/SchoolSubjectServiceImpl.java

```diff
@@ -33,44 +40,22 @@ public SchoolSubject saveOrUpdateSubject(SchoolSubject schoolSubject) {
         return subjectRepository.save(schoolSubject);
     }
 
-    @Override
-    public void deleteSubjectFromTeacher(Optional<Teacher> teacher, SchoolSubject schoolSubject) {
-
-        Set<SchoolSubject> subjects = teacher.get().getSubjects();
-        subjects.remove(schoolSubject);
-
-        teacherRepository.save(teacher.get());
-
-    }
-
-    @Override
-    public void deleteSubjectFromAllTeachers(Optional<List<Teacher>> teachers, SchoolSubject schoolSubject) {
-
-        for (Teacher teacher: teachers.get()) {
-            if (teacher.getSubjects().contains(schoolSubject)){
-                deleteSubjectFromTeacher(Optional.ofNullable(teacher), schoolSubject);
-            }
-        }
-
-        deleteSubject(schoolSubject.getId());
-
-    }
 
     @Override
     @Transactional
     public void deleteSubject(Long id) {
 
-        List<Teacher> teachers = teacherRepository.findAll();
+        List<Teacher> teachers = teacherService.findAllTeachers();
 
         SchoolSubject subject = subjectRepository.getOne(id);
 
         for (Teacher teacher: teachers) {
             if (teacher.getSubjects().contains(subject)){
                 teacher.getSubjects().remove(subject);
-                teacherRepository.save(teacher);
+                teacherService.saveOrUpdateTeacher(teacher);
             }
         }
 
-        subjectRepository.deleteById(id);
+        subjectRepository.delete(subject);
     }
```

Comment:
looks like you've missed findAllByOrderByName(), have you? it's very useful when you've got 12 grade levels in your school